ProcessLib/SmallDeformation/SmallDeformationFEM.h

```diff
@@ -0,0 +1,356 @@
+    using ModulusMatrixType = typename BMatricesType::ModulusMatrixType;
+
+    BMatrixType _b_matrices;
+    StressVectorType _sigma, _sigma_prev;
+    StrainVectorType _eps, _eps_prev;
+
+    Solids::SolidsBase<DisplacementDim>& _solid_material;
+    std::unique_ptr<
+        typename Solids::SolidsBase<DisplacementDim>::InternalVariables>
+        _internal_variables;
+
+    ModulusMatrixType _C;
+    double _detJ;
+
+    void pushBackState()
+    {
+        _eps_prev = _eps;
+        _sigma_prev = _sigma;
+        _internal_variables->pushBackState();
+    }
+};
+
+namespace ProcessLib
+{
+namespace SmallDeformation
+{
+enum class IntegrationPointValue
+{
+    SigmaXX,
+    SigmaYY,
+    SigmaXY,
+};
+
+template <typename ShapeMatrixType>
+struct SecondaryData
+{
+    std::vector<ShapeMatrixType> N;
+    std::vector<double> _sigmaXX;
+    std::vector<double> _sigmaYY;
+    std::vector<double> _sigmaXY;
+};
+
+struct SmallDeformationLocalAssemblerInterface
+    : public ProcessLib::LocalAssemblerInterface,
+      public NumLib::ExtrapolatableElement
+{
+    virtual std::vector<double> const& getIntPtSigmaXX(
+        std::vector<double>& /*cache*/) const = 0;
+
+    virtual std::vector<double> const& getIntPtSigmaYY(
+        std::vector<double>& /*cache*/) const = 0;
+
+    virtual std::vector<double> const& getIntPtSigmaXY(
+        std::vector<double>& /*cache*/) const = 0;
+};
+
+template <typename ShapeFunction, typename IntegrationMethod,
+          unsigned DisplacementDim>
+class SmallDeformationLocalAssembler
+    : public SmallDeformationLocalAssemblerInterface
+{
+public:
+    using ShapeMatricesType =
+        ShapeMatrixPolicyType<ShapeFunction, DisplacementDim>;
+    using NodalMatrixType = typename ShapeMatricesType::NodalMatrixType;
+    using NodalVectorType = typename ShapeMatricesType::NodalVectorType;
+    using ShapeMatrices = typename ShapeMatricesType::ShapeMatrices;
+    using BMatricesType = BMatrixPolicyType<ShapeFunction, DisplacementDim>;
+
+    using BMatrixType = typename BMatricesType::BMatrixType;
+    using StrainVectorType = typename BMatricesType::StressVectorType;
+    using StiffnessMatrixType = typename BMatricesType::StiffnessMatrixType;
+    using NodalForceVectorType = typename BMatricesType::NodalForceVectorType;
+    using NodalDisplacementVectorType =
+        typename BMatricesType::NodalForceVectorType;
+
+    SmallDeformationLocalAssembler(SmallDeformationLocalAssembler const&) =
+        delete;
+    SmallDeformationLocalAssembler(SmallDeformationLocalAssembler&&) = delete;
+
+    SmallDeformationLocalAssembler(
+        MeshLib::Element const& e,
+        std::size_t const local_matrix_size,
+        unsigned const integration_order,
+        SmallDeformationProcessData<DisplacementDim>& process_data)
+        : _process_data(process_data),
+          _localA(local_matrix_size, local_matrix_size),
+          _localRhs(local_matrix_size),
```

Comment:
Just wondering: I thought, you always assemble the residuum and the Jacobian...
But no rename needed. The variables will be gone after the new Jacobian assembly.